javascript/ql/test/query-tests/Security/CWE-022/TaintedWindowsPath/TaintedWindowsPath.qlref

```diff
@@ -0,0 +1 @@
+Security/CWE-022/TaintedWindowsPath.ql
```

Comment:
I would like a test of which taint-flows are only an issue on Windows.  

I've written it for you here:  
```CodeQL
import javascript
import semmle.javascript.security.dataflow.TaintedPathQuery

// TODO: Move this to TaintedWindowsPathQuery.qll?
class Win32Platform extends Path::Platform {
  Win32Platform() { this = Path::platformWin32() }

  override string getSep() { result = "\\" }
}

from Configuration cfg, DataFlow::PathNode source, DataFlow::PathNode sink
where
  cfg.hasFlowPath(source, sink) and
  sink.getFlowLabel().(Path::PathLabel).getPlatform() = Path::platformWin32() and
  not sink.getFlowLabel().(Path::PathLabel).getPlatform() = Path::platformPosix()
select source, sink
```

It detects that `express.js:27` is only an issue on windows (as expected).  
But it doesn't flag `express.js:35`. I haven't looked into why.